vm/src/obj/objstr.rs

```diff
@@ -518,6 +518,18 @@ impl PyString {
         do_cformat(vm, format_string, values.clone())
     }
 
+    #[pymethod(name = "__rmod__")]
+    fn rmod(&self, values: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        let format_string_text = &self.value;
+        let _format_string = CFormatString::from_str(format_string_text)
+            .map_err(|err| vm.new_value_error(format!("{}", err)))?;
+
+        match do_cformat(vm, _format_string, values.clone()) {
+            Ok(_format_string) => Ok(vm.ctx.not_implemented()),
```

Comment:
Is the call to `do_cformat` required here? Or can we just always return `NotImplemented`?

Reply:
I have just checked your second suggestion, and I found it works.
I used do_cformat just for argument checking, but that method is not required.
I will add a new commit for this PR.